services/datalad/datalad_service/handlers/files.py

```diff
@@ -131,6 +132,24 @@ def on_delete(self, req, resp, dataset, filename):
             else:
                 resp.media = {'error': 'no such file'}
                 resp.status = falcon.HTTP_NOT_FOUND
+        elif req.content_length:
+            filenames = json.loads(req.bounded_stream.read().decode('utf8'))['filenames']
+            ds_path = self.store.get_dataset_path(dataset)
+            deleted_files = [
+                filename for filename in filenames 
+                if os.path.exists(os.path.join(ds_path, filename))
+            ]
+            media_dict = {'deleted': deleted_files}
+            name, email = get_user_info(req)
+            if name and email:
+                media_dict['name'] = name
+                media_dict['email'] = email
+            try:
+                remove_files(self.store, dataset, files=deleted_files, name=name, email=email, cookies=req.cookies)
+                resp.media = media_dict
+                resp.status = falcon.HTTP_OK
+            except:
+                resp.status = falcon.HTTP_INTERNAL_SERVER_ERROR
```

Comment:
The implementation here looks fine but I wouldn't overload the same API now three paths to doing the same thing. If this is its own resource handler it wouldn't need the req.bounded_stream hack. req.media does this internally when accessed, including the JSON parsing.

This should also have test coverage.